File: Tests/CFLocale/create.c
#include <stdio.h>

#include <CoreFoundation/CFString.h>
#include <CoreFoundation/CFLocale.h>

#include "../Testing.h"

int main (void)
{
  CFLocaleRef locale;
  CFLocaleRef locale2;
  CFStringRef str;
  
  locale = CFLocaleCopyCurrent ();
  PASS_NEQ (locale, NULL, "CFLocaleCopyCurrent() returns a value");
  CFRelease((CFTypeRef)locale);
  
  locale = CFLocaleGetSystem ();
  PASS_NEQ (locale, NULL, "CFLocaleCopyCurrent() returns a value");
  
  locale = CFLocaleCreate (NULL, CFSTR("es_ES_PREEURO"));
  PASS_NEQ (locale, NULL, "CFLocaleCreate() returns a value");
  PASS_EQ (CFLocaleGetIdentifier(locale), CFSTR("es_ES@currency=ESP"),
    "en_ES_PREEURO is stored as es_ES@currency=ESP");
  locale2 = CFLocaleCreateCopy (NULL, locale);
  PASS_NEQ (locale2, NULL, "CFLocaleCreateCopy() returns a value");
  CFRelease((CFTypeRef)locale);
  CFRelease((CFTypeRef)locale2);
  
  return 0;
}

File: Tests/CFNumberFormatter/create.m
#include "CoreFoundation/CFString.h"
#include "CoreFoundation/CFLocale.h"
#include "CoreFoundation/CFNumberFormatter.h"
#include "../CFTesting.h"

int main (void)
{
  CFNumberFormatterRef fmt;
  CFLocaleRef locale;
  CFStringRef str;
  CFStringRef exp;
  
  locale = CFLocaleCreate (NULL, CFSTR("en_GB"));
  fmt = CFNumberFormatterCreate (NULL, locale, kCFNumberFormatterNoStyle);
  PASS_CF(fmt != NULL, "CFNumberFormatterCreate() return non-nil");
  str = CFNumberFormatterCopyProperty (fmt,
    kCFNumberFormatterDefaultFormat);
  PASS_CFEQ(str, CFSTR("#"), "Default no-style format same as cocoa");
  CFRelease (fmt);
  
  fmt = CFNumberFormatterCreate (NULL, locale,
    kCFNumberFormatterDecimalStyle);
  str = CFNumberFormatterCopyProperty (fmt,
    kCFNumberFormatterDefaultFormat);
  PASS_CFEQ(str, CFSTR("#,##0.###"),
    "Default decimal-style format same as cocoa");
  CFRelease (fmt);
  
  fmt = CFNumberFormatterCreate (NULL, locale,
    kCFNumberFormatterCurrencyStyle);
  str = CFNumberFormatterCopyProperty (fmt,
    kCFNumberFormatterDefaultFormat);
  exp = CFStringCreateWithCString (NULL, "¤#,##0.00", kCFStringEncodingUTF8);
  PASS_CFEQ(str, exp, "Default currency-style format same as cocoa");
  CFRelease (exp);
  CFRelease (fmt);
  
  fmt = CFNumberFormatterCreate (NULL, locale,
    kCFNumberFormatterPercentStyle);
  str = CFNumberFormatterCopyProperty (fmt,
    kCFNumberFormatterDefaultFormat);
  PASS_CFEQ(str, CFSTR("#,##0%"),
    "Default percent-style format same as cocoa");
  CFRelease (fmt);
  
  fmt = CFNumberFormatterCreate (NULL, locale,
    kCFNumberFormatterScientificStyle);
  str = CFNumberFormatterCopyProperty (fmt,
    kCFNumberFormatterDefaultFormat);
  PASS_CFEQ(str, CFSTR("#E0"),
    "Default scientific-style format same as cocoa");
  CFRelease (fmt);
  
  fmt = CFNumberFormatterCreate (NULL, NULL, kCFNumberFormatterSpellOutStyle);
  PASS_CF(fmt != NULL,
    "CFNumberFormatterCreate() return non-nil for system locale");
  
  CFRelease ((CFTypeRef)locale);
  CFRelease (fmt);
  
  return 0;
}
